core/indexer/common.go

```diff
@@ -448,15 +412,62 @@ func prepareTxUpdate(tx *Transaction) ([]byte, []byte) {
 
 	if tx.GasUsed == tx.GasLimit {
 		// do not update gasUsed because it is the same with gasUsed when transaction was saved first time in database
-		serializedData = []byte(fmt.Sprintf(`{ "doc" : { "log" : %s, "scResults" : %s, "status": "%s", "timestamp": %s } }`,
-			string(marshalizedLog), string(scResults), tx.Status, string(marshalizedTimestamp)))
+		serializedData =
+			[]byte(fmt.Sprintf(`{"script":{"source":"`+
+				`ctx._source.status = params.status;`+
+				`ctx._source.log = params.log;`+
+				`ctx._source.scResults = params.scResults;`+
+				`ctx._source.timestamp = params.timestamp;`+
+				`","lang": "painless","params":`+
+				`{"status": "%s", "log": %s, "scResults": %s, "timestamp": %s}},"upsert":%s}`,
+				tx.Status, string(marshalizedLog), string(scResults), string(marshalizedTimestamp), string(marshalledTx)))
 	} else {
 		// update gasUsed because was changed (is a smart contract operation)
-		serializedData = []byte(fmt.Sprintf(`{ "doc" : { "log" : %s, "scResults" : %s, "status": "%s", "timestamp": %s, "gasUsed" : %s } }`,
-			string(marshalizedLog), string(scResults), tx.Status, string(marshalizedTimestamp), fmt.Sprintf("%d", tx.GasUsed)))
+		serializedData =
+			[]byte(fmt.Sprintf(`{"script":{"source":"`+
+				`ctx._source.status = params.status;`+
+				`ctx._source.log = params.log;`+
+				`ctx._source.scResults = params.scResults;`+
+				`ctx._source.timestamp = params.timestamp;`+
+				`ctx._source.gasUsed = params.gasUsed;`+
+				`","lang": "painless","params":`+
+				`{"status": "%s", "log": %s, "scResults": %s, "timestamp": %s, "gasUsed": %d}},"upsert":%s}`,
+				tx.Status, string(marshalizedLog), string(scResults), string(marshalizedTimestamp), tx.GasUsed, string(marshalledTx)))
+	}
+
+	log.Warn("log meta - update", "data", string(meta))
+	log.Warn("log srld data - update", "data", string(serializedData))
+	//log.Debug("index tx request", "request", string(serializedData))
+
+	return
+}
+
+func prepareSerializedAccountInfo(address string, account *AccountInfo) (meta []byte, serializedData []byte) {
+	var err error
+	meta = []byte(fmt.Sprintf(`{ "index" : { "_id" : "%s" } }%s`, address, "\n"))
+	serializedData, err = json.Marshal(account)
+	if err != nil {
+		log.Debug("indexer: marshal",
+			"error", "could not serialize account, will skip indexing",
+			"address", address)
+		return
 	}
 
-	return meta, serializedData
+	return
+}
+
+func prepareSerializedAccountBalanceHistory(address string, account *AccountBalanceHistory) (meta []byte, serializedData []byte) {
+	var err error
+	meta = []byte(fmt.Sprintf(`{ "index" : { "_id" : "%s" } }%s`, address, "\n"))
+	serializedData, err = json.Marshal(account)
+	if err != nil {
+		log.Debug("indexer: marshal",
```

Comment:
here also, do not skip indexing when an error occurs.

Reply:
done. with a note: the errors for specified areas will now arrive in the `workItem` part that won't stop the node's execution as I know.. so we will need another task maybe for treating differently the errors there

Reply:
it will eventually stop the node as the queue will not be consumed (if the item.Save errors - by any mean - it will get stuck in  a retry mode) and that queue will fill up, preventing syncBlock to finish executing.

Reply:
We should definitely decide how to treat this kind of errors.
In this specific case we should treat it as an invalid document and skip it. I am saying this because if it's a `json.Marshal` error, the node will crash anyway.